src/pointings_search/pointing_table.py

```diff
@@ -62,36 +61,51 @@ def from_csv(cls, filename, alt_names=None):
         return result
 
     @classmethod
-    def from_sqllite(cls, db_name, table_name, columns_map):
+    def from_sqlite(cls, db_name, table_name, columns_map=None):
         """Create the PointingTable from a sqllite database file.
 
-        The columns map must have the following
-
         Parameters
         ----------
         db_name : `str`
             The file location of pointing database.
         table_name : `str`
             The table to query from the database.
-        columns_map : `dict`
+        columns_map : `dict`, optional
             A dictionary indcating which columns to load and mapping the
             column name in the PointingTable to the column name in the SQL database.
             Must contain all required columns.
         """
+        # Open the database.
+        connection = sqlite3.connect(db_name)
+
+        # If we do not have a column map, load the entire table with the existing column names.
+        if columns_map is None:
+            columns_map = {}
+            colmap_cursor = connection.cursor()
+            select_data = colmap_cursor.execute(f"SELECT * FROM {table_name} LIMIT 1")
+            for col in select_data.description:
+                columns_map[col[0]] = col[0]
+
+        # Construct both the SQL query and the dictionary to hold the results.
+        data_dict = {}
         db_query = "SELECT"
         for key in columns_map:
+            data_dict[key] = []
             if len(db_query) <= 8:
                 db_query += f" {columns_map[key]} as {key}"
             else:
                 db_query += f", {columns_map[key]} as {key}"
         db_query += f" FROM {table_name}"
 
         # Read the data from the database.
-        connection = sqlite3.connect(db_name)
-        pandas_df = pd.read_sql_query(db_query, connection)
-
-        # Convert the data into an astropy Table (from the Panda's table).
-        ap_table = Table.from_pandas(pandas_df)
+        row_cursor = connection.cursor()
+        for row in row_cursor.execute(db_query):
+            for i, key in enumerate(data_dict):
+                data_dict[key].append(row[i])
+        connection.close()
```

Comment:
I'm happy that the new code closes the connection, but I'd feel better if this did one of those `with` context things, just so the connection close gets called even if there's an error encountered in the intervening code.

Reply:
Done (here and below)